components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2042,9 +2042,30 @@ public int compare(AuthenticationStep step1, AuthenticationStep step2) {
 
             Arrays.sort(authenticationSteps, comparator);
 
+            int numSteps = authenticationSteps.length;
+            // We check if the steps have consecutive step numbers
+            if (numSteps > 0 && authenticationSteps[numSteps-1].getStepOrder() != numSteps) {
+                // iterate through the steps and fix step order
+                int count = 1;
+                for (AuthenticationStep step : authenticationSteps) {
+                    step.setStepOrder(count++);
```

Comment:
You want to start step number from 2 ?

Reply:
No, count++ will have the counter incremented once line 2051 is executed. So in the first iteration count will be 1